aws-frauddetector-detector/src/aws_frauddetector_detector/helpers/model_helpers.py

```diff
@@ -105,6 +104,11 @@ def get_model_for_detector(frauddetector_client, detector, model: models.Resourc
             # we should never see this block get executed
             raise exceptions.NotFound("associatedModel", model_endpoint)
         associated_models.append(models.Model(Arn=external_models[0].get("arn", "not/found")))
+
+    model_versions: List[dict] = desired_detector_version.get("modelVersions", [])
+    for model_version in model_versions:
+        associated_models.append(models.Model(Arn=model_version["arn"]))
```

Comment:
Sanity check: this will fetch current state of events for model version(s) attached to Detector Version. This will cover out of band updates to this Detector?

Reply:
If the customer attaches a different model version to this detector outside of CFN, will the next read from CFN will have that new model version's arn as an AssociatedModel, yes.

One comment here, if for some reason, AFD does not return `arn` for the model version, we would get KeyError on line 110. Thoughts on avoiding a KeyError for `Arn=model_version["arn"]`?

Reply:
We could definitely get a `KeyError` here, but that would be a highly unexpected scenario given that we expect an array of ModelVersions from `get_detector_version`, and if that array is non empty, it should contain an ARN.

It does not hurt to add a check here for key error, and add some context that we do not expect that line to execute in the comments.